### PR TITLE
Aligning `Pool.ts` with `CLPool.ts` modules

### PR DESCRIPTION
**Pull request by FonsSicca**
Implements:
- Alignment of code pattern of `Pool.ts` with that of `CLPool.ts`

## Summary by CodeRabbit

* **Bug Fixes**
  * More robust handling when pools or user data are missing to avoid partial updates.

* **Improvements**
  * Enhanced USD-based calculations for fees, liquidity and swap volumes.
  * More accurate timestamp and block rounding during pool data refresh for consistent history.

* **Tests**
  * Expanded test coverage for pool event edge cases, OUSDT swap scenarios, and updated swap/fee/liquidity behaviors.

<sub>✏️ Tip: You can customize this high-level summary in your review settings.</sub>

### CODE REVIEW

File: src/EventHandlers/Pool/PoolBurnAndMintLogic.ts [hunk 1]
```diff
@@ -16,62 +15,59 @@ export interface PoolLiquidityResult {
...
   token1Instance: Token,
   amount0: bigint,
   amount1: bigint,
-  context: handlerContext,
-): Promise<PoolLiquidityResult> {
-  // Update reserve data
-  const reserveData = await updateReserveTokenData(
-    token0Instance,
-    token1Instance,
+): PoolLiquidityResult {
```
[the remark this answers is not in the document]
Comment (FonsSicca) [reply src/EventHandlers/Pool/PoolBurnAndMintLogic.ts]: assessed already in next PR (didn't do in this one because the review of this PR was done after the next PR was pushed)

**Comment by FonsSicca** [comment]
@coderabbitai review

**Review by WooSungD — APPROVED** [review]
